### PR TITLE
Hook F.cross_entropy for nntile tensors

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Routes `torch.nn.functional.cross_entropy` to the existing nntile cross-entropy implementation when logits are on `device="nntile"`, mirroring the `F.rms_norm` hook in `normalization.py`.

## Changes

- Add [`torch_nntile/torch_nntile/loss.py`](torch_nntile/torch_nntile/loss.py) with `cross_entropy()` wrapper and `patch_cross_entropy()`
- Import `loss` from [`torch_nntile/__init__.py`](torch_nntile/torch_nntile/__init__.py) so the patch applies on `import torch_nntile`
- Resolve deprecated `size_average` / `reduce` via `_Reduction.legacy_get_string` on the nntile path before validation (Bugbot fix)

## Behavior

- **CPU/CUDA tensors:** unchanged — delegates to the original `F.cross_entropy`
- **Nntile tensors:** delegates to existing `training.cross_entropy` / `_NntileCrossEntropy` / C++ path
- **Legacy args:** `size_average` / `reduce` are resolved the same way as native `F.cross_entropy` before nntile validation
- **Unsupported on nntile:** `weight`, `label_smoothing`, `reduction='none'` raise `ValueError` (same limits as before)

## Usage

```python
import torch
import torch.nn.functional as F
import torch_nntile

logits = model(x)  # device="nntile"
loss = F.cross_entropy(logits, targets, reduction="mean", ignore_index=-100)
```

## Out of scope

No C++, executor, test, example, or `train_full_batch_step` changes. `training.cross_entropy` remains available as the internal implementation.

<div><a href="https://cursor.com/agents/bc-30e8b512-5be2-4112-8c4f-4b2020b9cc56"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-30e8b512-5be2-4112-8c4f-4b2020b9cc56"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@bugbot review
